engine/utils/parseFilters.go

```diff
@@ -0,0 +1,41 @@
+package utils
+
+import (
+	"fmt"
+	"reflect"
+	"strings"
+)
+
+// FilterMapByStructFields returns a new map containing only the fields that exist in the provided struct.
+func FilterMapByStructFields(fieldMap map[string]interface{}, filtersStruct interface{}) (map[string]interface{}, error) {
```

Comment:
It looks like the `FilterMapByStructFields` function is no longer being used after my changes. Could you please remove it?

It would also be great if you could take a quick look for any other functions that might not be needed anymore.